device/virtual/x86_64-kvm_x86_64-r0/virtual_chassis/virtual_chassis.py

```diff
@@ -0,0 +1,293 @@
+   res = ret.read().strip()
+   if res != "":
+      print cmd
+      print res
+   #print res
+   return res
+
+
+class VirtualChassisTopology(object):
+   def __init__(self, topoFile, oper, ns, src=None, image=None):
+      self.instInfo = {}
+      self.topoFile = topoFile
+      self.oper = oper
+      self.ns = ns
+      self.src = src
+      self.image = image
+      self.nss = "sudo ip netns exec " + self.ns
+      self.brReq = {}
+      self.readyIntfsInInst = {}
+      self.chassBr = "br4chs"
+      self.ctns = {}
+
+   def handleRequest(self):
+      with open(self.topoFile, 'r') as f:
+         virtualTopology = json.load(f)["VIRTUAL_TOPOLOGY"]
+         inst = virtualTopology['chassis_instances']
+         if 'neighbor_instances' in virtualTopology:
+            inst += virtualTopology['neighbor_instances']
+         if self.oper == 'verify':
+            self.verifyVirtualChassis(virtualTopology)
+            return
+         if self.oper == 'create':
+            runCmd("sudo ip netns add " + self.ns)
+            self.handleBridge(self.chassBr)
+            # docker-sonic-vs requires eth0 to be present for orchagent
+            for i in inst:
+               self.brReq[i] = [("eth0", "")]
+               self.readyIntfsInInst[i] = []
+
+            if 'neighbor_connections' in virtualTopology:
+               for nck, ncv in virtualTopology['neighbor_connections'].items():
+                  for i in ncv:
+                     self.brReq[i].append((ncv[i], "br" + nck))
+         for instDir in inst:
+            self.handleInstance(instDir)
+         if 'neighbor_connections' in virtualTopology:
+            for nck, ncv in virtualTopology['neighbor_connections'].items():
+               self.handleNeighConn(nck, ncv)
+         if self.oper == 'delete':
+            self.handleBridge("br4chs")
+            runCmd("sudo ip netns del " + self.ns)
+
+   def handleBridge(self, brName):
+      if self.oper == "create":
+         runCmd(self.nss + " brctl addbr " + brName)
+         runCmd(self.nss + " ip link set dev " + brName + " up")
+      else:
+         runCmd(self.nss + " ip link set dev " + brName + " down")
+         runCmd(self.nss + " brctl delbr " + brName)
+      return
+
+   def setInstInfo(self, inst, name, pid):
+      self.instInfo[inst] = [name, pid]
+
+   def getInstInfo(self, inst):
+      res = self.instInfo[inst]
+      return res[0], res[1]
+
+   def getContainer(self, hostName):
+      if hostName in self.ctns:
+         return self.ctns[hostName]
+      for ctn in docker.from_env().containers.list():
+         if ctn.name == hostName:
+            self.ctns[ctn.name] = ctn
+            return ctn
+      return None
+
+   def runCmdOnHost(self, hostName, cmd):
+      #print hostName + " " + cmd
+      ctn = self.getContainer(hostName)
+      res = ctn.exec_run(cmd)
+      try:
+         out = res.output
+      except AttributeError:
+         out = res
+      return out
+
+   def handleInstance(self, instDir):
```

Comment:
I'm still not 100% sure what "instance" is referring to after looking over this. Linecard? Supervisor? Neighbor? All of the above? I'd call this `configureSonicInstance` or something to make it clear to other contributors what this is supposed to be.

I would also consider breaking the create step up into a "create the container" helper and a "create the network" helper as well.

Reply:
instance is referring to any of the docker-sonic-vs instances. it can be linecard/supervisor/neighbor. 
I can rename handleInstance to handleSonicInstance as it does handle both create/delete operations with docker create or docker stop/rm

Reply:
we will address this in PR2.

Reply:
we will address this in PR2.